### PR TITLE
feat: migrate some components to typescript

### PR DESCRIPTION
**Pull request by chambo-e**
Image, Label, PhoneInput, ProgressBar, ProgressionButton and StateBar

### CODE REVIEW

File: src/components/Image/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types'
+import React, { FunctionComponent } from 'react'
+import Box from '../Box'
+
+const Image: FunctionComponent<HTMLImageElement & XStyledProps> = ({
+  src,
+  ...props
+  // @ts-expect-error Box expose Element, not ImgElement
+}) => <Box as="img" src={src} {...props} />
```
Comment (adriengibrat) [inline src/components/Image/index.tsx:9]: Props are `HTMLAttributes<HTMLImageElement>` & you may just forward all props:
```suggestion
const Image: FunctionComponent<HTMLAttributes<HTMLImageElement> & XStyledProps> = (props) => <Box as="img" src={src} {...props} />
```

File: src/components/Label/index.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const StyledLabel = styled.label`
+  color: ${({ theme }) => theme.colors.gray950};
+  font-size: 16px;
+  line-height: 24px;
+  font-weight: 500;
+  margin-bottom: 8px;
+`
+
+const Label: FunctionComponent = props => <StyledLabel {...props} />
```
Comment (adriengibrat) [inline src/components/Label/index.tsx:12]: Why not just:

```suggestion
import React from 'react'

const Label = styled.label`
  color: ${({ theme }) => theme.colors.gray950};
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  margin-bottom: 8px;
`
```

Comment (chambo-e) [reply src/components/Label/index.tsx:12]: This is a workaround to have ArgTable working in Storybook, not that pretty yeah ...

File: src/components/Image/index.tsx (hunk 1 above)
Comment (chambo-e) [reply src/components/Image/index.tsx]: Corrected thanks :)

Comment (chambo-e) [reply src/components/Image/index.tsx]: Corrected thanks :)

**Review by DorianMaliszewski — APPROVED** [review]

File: src/components/Label/index.tsx (hunk 2 above)
Comment (adriengibrat) [inline src/components/Label/index.tsx:12]:
```suggestion
const Label: FunctionComponent<LabelHTMLAttributes<HTMLLabelElement>> = props => <StyledLabel {...props} />
```

Comment (chambo-e) [reply src/components/Label/index.tsx:12]: ffs, that component proxy really stinks ...

File: src/components/PhoneInput/index.tsx [hunk 3]
```diff
@@ -87,15 +97,33 @@ const StyledInput = styled.input`
...
+  disableDropdown = false,
+  inputProps: { name, id, placeholder, 'data-testid': dataTestId } = {},
   onChange,
   value,
-  label,
+  label = 'Phone',
 }) => {
-  const inputRef = useRef()
+  const inputRef =
+    useRef<HTMLInputElement>() as React.MutableRefObject<HTMLInputElement>
```
Comment (vincentaudebert) [inline src/components/PhoneInput/index.tsx:126]: importer pour eviter `React.` ?

File: src/components/PhoneInput/index.tsx [hunk 4]
```diff
@@ -87,15 +97,33 @@ const StyledInput = styled.input`
...
+    iti: intlTelInput.Plugin,
+  ) => void
+  value?: string
+  label?: string
+}
+
+const PhoneInput: VoidFunctionComponent<PhoneInputProps> = ({
+  disabled = false,
+  disableDropdown = false,
+  inputProps: { name, id, placeholder, 'data-testid': dataTestId } = {},
```
Comment (vincentaudebert) [inline src/components/PhoneInput/index.tsx:120]: on peut pas definir les valeurs par defaut ici pour eviter les `?? undefined` plus bas?

File: src/components/Pentagon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  clip-path: polygon(50% 0%, 100% 38%, 82% 100%, 18% 100%, 0% 38%);
+
+  ${({ size = '48px', color = 'pippin', theme }) => `
+    width: ${size};
+    height: ${size};
+    background-color: ${theme.colors[color as Color] ?? color};
+  `}
+`
+
+const Pentagon: FunctionComponent<PentagonProps> = props => (
```
Comment (adriengibrat) [inline src/components/Pentagon/index.tsx:26]: Either
```suggestion
const Pentagon: FunctionComponent<PentagonProps & HTMLAttributes<HTMLDivElement>> = props => (
```
or
```suggestion
const Pentagon: FunctionComponent<PentagonProps> = ({size, color}) => (
```

File: src/components/PhoneInput/index.tsx [hunk 6]
```diff
@@ -87,15 +97,33 @@ const StyledInput = styled.input`
   }
 `
 
-const PhoneInput = ({
-  disabled,
-  disableDropdown,
-  inputProps: { name, id, placeholder, ...inputProps },
+interface InputProps {
```
Comment (adriengibrat) [inline src/components/PhoneInput/index.tsx:100]:
```suggestion
type InputProps = InputHTMLAttributes<HTMLInputElement>
```

File: src/components/PhoneInput/index.tsx [hunk 7]
```diff
@@ -144,12 +173,12 @@ const PhoneInput = ({
           type="tel"
           ref={inputRef}
           value={value}
-          name={name}
-          id={id}
+          name={name ?? undefined}
```
Comment (adriengibrat) [inline src/components/PhoneInput/index.tsx:176]: :crying_cat_face:

File: src/components/ProgressBar/index.tsx [hunk 8]
```diff
@@ -17,15 +18,15 @@ const shineAnimation = keyframes`
 export const progressBarVariants = ['primary', 'success', 'warning', 'info']
 
 const StyledBox = styled(Box, {
-  shouldForwardProp: prop => !['backgroundColor'].includes(prop),
-})`
+  shouldForwardProp: prop => !['backgroundColor'].includes(prop.toString()),
```
Comment (adriengibrat) [inline src/components/ProgressBar/index.tsx:21]: simpler & efficient (but less extensible) suggestion:
```suggestion
  shouldForwardProp: prop => 'backgroundColor' === prop,
```

File: src/components/ProgressBar/index.tsx [hunk 9]
```diff
@@ -17,15 +18,15 @@ const shineAnimation = keyframes`
...
+})<{ backgroundColor: string }>`
```
Comment (adriengibrat) [inline src/components/ProgressBar/index.tsx:22]: Why not type `backgroundColor` as `Color` all the way down ?

```suggestion
})<{ backgroundColor?: Color }>`
```

File: src/components/PhoneInput/index.tsx (hunk 4 above)
Comment (chambo-e) [reply src/components/PhoneInput/index.tsx:121]: fixed

File: src/components/PhoneInput/index.tsx (hunk 3 above)
Comment (chambo-e) [reply src/components/PhoneInput/index.tsx]: fixed

File: src/components/ProgressBar/index.tsx (hunk 8 above)
Comment (chambo-e) [reply src/components/ProgressBar/index.tsx:21]: This is the convention we have elsewhere, we can keep it that way to be consistent :)

**Review by philibea — APPROVED** [review]

**Review by vincentaudebert — APPROVED** [review]

**Pull request merged**